src/Symfony/Component/Ldap/LdapClient.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+namespace Symfony\Component\Ldap;
+
+use Symfony\Component\Ldap\Exception\ConnectionException;
+use Symfony\Component\Ldap\Exception\LdapException;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Francis Besset <francis.besset@gmail.com>
+ * @author Charles Sarrazin <charles@sarraz.in>
+ */
+class LdapClient implements LdapClientInterface
+{
+    private $host;
+    private $port;
+    private $version;
+    private $useSsl;
+    private $useStartTls;
+    private $optReferrals;
+    private $connection;
+    private $charmaps;
+
+    /**
+     * Constructor.
+     *
+     * @param string $host
+     * @param int    $port
+     * @param int    $version
+     * @param bool   $useSsl
+     * @param bool   $useStartTls
+     * @param bool   $optReferrals
+     */
+    public function __construct($host = null, $port = 389, $version = 3, $useSsl = false, $useStartTls = false, $optReferrals = false)
```

Comment:
It's impossible to specify additional options, such as timeout, debug level, etc... As this component should be used as a standalone, I suggest to put options here or make an interface to adjust each option, available for `ldap_set_option`

Example:
```php
$ldap = new LdapClient();
$ldap->setNetworkTimeout(10);
$ldap->setDebugLevel(7);
$ldap->setUseSsl(true);
$ldap->setProtocolVersion(3);
```